tools/build-benchmark.cmd

```diff
@@ -0,0 +1,35 @@
+@echo off
+set VS_TOOLS_VERSION=vs2019
+set CMAKE_GEN="Visual Studio 16 2019"
+echo Building Google Benchmark (test only dependency)...
+@setlocal ENABLEEXTENSIONS
+
+echo Auto-detecting Visual Studio version...
+call "%~dp0\vcvars.cmd"
+
+pushd "%~dp0\.."
+set "ROOT=%CD%"
+
+set MAXCPUCOUNT=%NUMBER_OF_PROCESSORS%
+set platform=
+
+if not exist "%ROOT%\third_party\benchmark\" (
+  echo "Google Benchmark library is not available, skipping benchmark build."
+  call skip_the_build
+)
+
+cd "%ROOT%\third_party\benchmark\"
+set "GOOGLETEST_PATH=%ROOT%\third_party\googletest"
+if not exist "build" (
+  mkdir build
+)
+cd build
+
+REM By default we generate the project for the older Visual Studio 2017 even if we have newer version installed
+cmake ../ -G %CMAKE_GEN% -Ax64 -DBENCHMARK_ENABLE_TESTING=OFF
+set SOLUTION=%ROOT%\third_party\benchmark\build\benchmark.sln
+msbuild %SOLUTION% /maxcpucount:%MAXCPUCOUNT% /p:Configuration=Debug /p:Platform=x64
```

Comment:
Use `cmake --build .` instead of calling `msbuild` to be consistent?

Reply:
We may be able to remove this script, as right now we have a few ways to inject the benchmark via `vcpkg`.

Reply:
I'll keep it as-is for now. But it's not used right now by the outside `build.cmd` script. I'll probably remove this `build-benchmark.cmd` in a follow-up PR.